pilot/pkg/networking/core/v1alpha3/tracing_test.go

```diff
@@ -45,80 +46,96 @@ func TestConfigureTracing(t *testing.T) {
-			wantRfCtx: nil,
+			name:            "both tracing enabled (with provider)",
+			inSpec:          fakeTracingSpec(fakeZipkin(), 99.999, false, true),
+			opts:            fakeOptsMeshAndTelemetryAPI(true /* enable tracing */),
+			want:            fakeTracingConfig(fakeZipkinProvider(clusterName, providerName), 99.999, 256, append(defaultTracingTags(), fakeEnvTag)),
+			wantRfCtx:       nil,
+			wantReqIDExtCtx: &defaultUUIDExtensionCtx,
 		},
 		{
-			name:      "both tracing disabled (with provider)",
-			inSpec:    fakeTracingSpec(fakeZipkin(), 99.999, false),
-			opts:      fakeOptsMeshAndTelemetryAPI(false /* no enable tracing */),
-			want:      fakeTracingConfig(fakeZipkinProvider(clusterName, providerName), 99.999, 256, append(defaultTracingTags(), fakeEnvTag)),
-			wantRfCtx: nil,
+			name:            "both tracing disabled (with provider)",
+			inSpec:          fakeTracingSpec(fakeZipkin(), 99.999, false, true),
+			opts:            fakeOptsMeshAndTelemetryAPI(false /* no enable tracing */),
+			want:            fakeTracingConfig(fakeZipkinProvider(clusterName, providerName), 99.999, 256, append(defaultTracingTags(), fakeEnvTag)),
+			wantRfCtx:       nil,
+			wantReqIDExtCtx: &defaultUUIDExtensionCtx,
 		},
 		{
-			name:      "basic config (with skywalking provicer)",
-			inSpec:    fakeTracingSpec(fakeSkywalking(), 99.999, false),
-			opts:      fakeOptsOnlySkywalkingTelemetryAPI(),
-			want:      fakeTracingConfig(fakeSkywalkingProvider(clusterName, providerName), 99.999, 0, append(defaultTracingTags(), fakeEnvTag)),
-			wantRfCtx: &xdsfilters.RouterFilterContext{StartChildSpan: true},
+			name:            "basic config (with skywalking provicer)",
+			inSpec:          fakeTracingSpec(fakeSkywalking(), 99.999, false, false),
+			opts:            fakeOptsOnlySkywalkingTelemetryAPI(),
+			want:            fakeTracingConfig(fakeSkywalkingProvider(clusterName, providerName), 99.999, 0, append(defaultTracingTags(), fakeEnvTag)),
+			wantRfCtx:       &xdsfilters.RouterFilterContext{StartChildSpan: true},
+			wantReqIDExtCtx: &requestidextension.UUIDRequestIDExtensionContext{UseRequestIDForTraceSampling: false},
 		},
 	}
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
 			hcm := &hpb.HttpConnectionManager{}
-			gotRfCtx := configureTracingFromSpec(tc.inSpec, tc.opts, hcm)
+			gotRfCtx, gotReqIDExtCtx := configureTracingFromSpec(tc.inSpec, tc.opts, hcm)
 			if diff := cmp.Diff(tc.want, hcm.Tracing, protocmp.Transform()); diff != "" {
 				t.Fatalf("configureTracing returned unexpected diff (-want +got):\n%s", diff)
 			}
 			if diff := cmp.Diff(gotRfCtx, tc.wantRfCtx); diff != "" {
 				t.Fatalf("got filter modifier context is unexpected diff (-want +got):\n%s", diff)
 			}
+			if diff := cmp.Diff(tc.wantReqIDExtCtx, gotReqIDExtCtx); diff != "" {
+				t.Fatalf("configureTracing returned unexpected diff (-want +got):\n%s", diff)
```

Comment:
nit: add something in the message about reqIDExtCtx, like `configureTracingFromSpec returned unexpected diff for request ID extension (-want +got):\n%s` or similar